spec/dynamoid/persistence_spec.rb

```diff
@@ -292,6 +292,32 @@
         msg.destroy
       end.to_not raise_error
     end
+
+    context 'with lock version' do
+      it 'deletes a record if lock version matches' do
+        address.save!
+        expect { address.destroy }.to_not raise_error
```

Comment:
The destroy method in ActiveRecord does not catch ActiveRecord::StaleObjectError, so its the responsibility of the caller to handle those exceptions.

https://github.com/rails/rails/blob/7f18ea14c893cb5c9f04d4fda9661126758332b5/activerecord/lib/active_record/persistence.rb#L179

I tried to keep the same format as ActiveRecord. Although I understand what you mean, as a reader it looks like destroy does not throw an exception, I can make the modification.